arch/arm64/boot/dts/broadcom/Makefile

```diff
@@ -1,7 +1,14 @@
+# Enable fixups to support overlays on BCM2835 platforms
+
+ifeq ($(CONFIG_ARCH_BCM2835),y)
+DTC_FLAGS ?= -@ -H epapr
+dts-dirs += ../overlays
+endif
+
 dtb-$(CONFIG_ARCH_BCM2835) += bcm2837-rpi-3-b.dtb
 dtb-$(CONFIG_ARCH_BCM_IPROC) += ns2-svk.dtb
 dtb-$(CONFIG_ARCH_VULCAN) += vulcan-eval.dtb
-dtb-$(CONFIG_ARCH_BCM2709) += bcm2710-rpi-3-b.dtb
+dtb-$(CONFIG_ARCH_BCM2835) += bcm2710-rpi-3-b.dtb
 
 always		:= $(dtb-y)
 subdir-y	:= $(dts-dirs)
```

Comment:
From what I have found, the clean rules are applied without regard for the configuration. This makes sense, since when you are changing configuration you want the old configuration to be cleaned, not the new one. Because overlays is only added to dts-dirs when ARCH_BCM2835 is set, this means that the .dtb files are never deleted.

`arch/arm/boot/dts/Makefile` has the following:
```
subdir-y        := overlays
clean-files     := *.dtb
```
I think you need at least the subdir-y change, and perhaps clean-files as well.